### ISSUE TITLE
Possible issues with the calculation of CBR

### ISSUE THREAD

**Issue by asdLeaving** [wsxtyrdd/NTSCC_plus#5]
In line 412, net/ntscc.py, the channel usage "k" is calculated as the sum of effective mask **divided by 2**, which is presumably calculating the **complex** symbols required to be transmitted.

    def feature_pass_channel(self, s_masked, mask=None):
        if mask is None:
            mask = torch.ones_like(s_masked)
        avg_pwr = torch.sum(s_masked ** 2) / mask.sum()
        s_hat, _ = self.channel.forward(s_masked, avg_pwr)
        s_hat = s_hat * mask
        channel_usage = mask.sum() / 2
        return s_hat, channel_usage

In line 32, main.py, "k" is directly divided by the number of elements of the image (m = N * C * H * W) to calculate CBR.

    def forward(self, output, target, num_pixels=None):
        N, C, H, W = target.size()
        out = {}
        if num_pixels is None:
            num_pixels = N * H * W
        out["bpp_y"] = torch.log(output["likelihoods"]["y"]).sum() / (-math.log(2) * num_pixels)
        out["bpp_z"] = torch.log(output["likelihoods"]["z"]).sum() / (-math.log(2) * num_pixels)
        out["cbr_y"] = output["k"] / (N * C * H * W)
        out["mse_loss_ntc"] = self.mse(output["x_hat_ntc"], target)
        out["mse_loss"] = self.mse(output["x_hat"], target)
        out["bpp_loss"] = out["bpp_y"] + out["bpp_z"]
        out["cbr"] = out["cbr_y"]

        out["ntscc_loss"] = self.lmbda * 255 ** 2 * out["mse_loss"] + out["bpp_loss"]
        out["loss"] = self.lmbda * 255 ** 2 * (out["mse_loss"] + out["mse_loss_ntc"]) + out["bpp_loss"]
        return out

However, the paper defines CBR as the number of **real** symbols divided by m, which is double the value calculated by the code. I wonder if other methods tested have the same issue (especially the traditional separate coding methods).

**Comment by wsxtyrdd** [comment]
To maintain consistency with previous works (for example, djscc), in practice, we convert two consecutive real symbols into one complex channel-input symbol and add complex Gaussian noise. 
In all experiments, we calculate the CBR using the number of **complex** symbols divided by the source dimensions.
I'm sorry for the confusion caused by the definitions in the paper.
